omas/omas_plot.py

```diff
@@ -359,21 +359,33 @@ def geo_type_lookup(geometry_type, subsys, imas_version=default_imas_version, re
 
     :return: string (or int if reverse=True)
         Name of the field indicated by geometry_type (or type code if reverse=True).
-        For example: In IMAS 3.19.0, `pf_active.coil[:].element[:].geometry.geometry_type = 0` means 'outline'.
+        For example: In IMAS 3.19.0, `pf_active.coil[:].element[:].geometry.geometry_type = 1` means 'outline'.
         In version 3.19.0 the following geometry types exist {1: 'outline', 2: 'rectangle', 4: 'arcs of circle'}
     """
 
-    # fetche information from IMAS data description of pf_active.coil[:].element[:].geometry.geometry_type
-    geo_map=eval('{%s}'%omas_info_node("pf_active.coil.:.element.:.geometry.geometry_type",
-                                       imas_version=imas_version)['documentation'].split('(')[-1][:-2])
-    print(geo_map)
+    # Fetch information from IMAS data description of geometry_type for the relevant subsys
+    lookup = {
+        'ic_antennas': 'ic_antennas.antenna.:.strap.:.geometry.geometry_type',
+        'pf_active': 'pf_active.coil.:.element.:.geometry.geometry_type',
+    }
+    if subsys not in lookup.keys():
+        print('Warning: unrecognized IMAS substructure ({})'.format(subsys))
+        return None
+
+    try:
+        doc = omas_info_node(lookup[subsys], imas_version=imas_version)['documentation']
+    except ValueError:
+        print('Warning: unrecognized IMAS version ({})'.format(imas_version))
```

Comment:
I'd suggest:

```python
except ValueError as _excp:
   printe(repr(_excp))
```

and then make sure that the error raised by an invalid `imas_version` is descriptive.